diff --git a/.pio/libdeps/pico/Adafruit TinyUSB Library/src/Adafruit_TinyUSB.h b/.pio/libdeps/pico/Adafruit TinyUSB Library/src/Adafruit_TinyUSB.h
index 1d0808c..25b5f95 100644
--- a/.pio/libdeps/pico/Adafruit TinyUSB Library/src/Adafruit_TinyUSB.h
+++ b/.pio/libdeps/pico/Adafruit TinyUSB Library/src/Adafruit_TinyUSB.h
@@ -25,6 +25,9 @@
 #ifndef ADAFRUIT_TINYUSB_H_
 #define ADAFRUIT_TINYUSB_H_
 
+#include <Arduino.h>
+extern HardwareSerial *TinyUSB_Serial_Debug;
+
 // Error message for Core that must select TinyUSB via menu
 #if !defined(USE_TINYUSB) && ( defined(ARDUINO_ARCH_SAMD) || \
                                (defined(ARDUINO_ARCH_RP2040) && !defined(ARDUINO_ARCH_MBED)) )
diff --git a/.pio/libdeps/pico/Adafruit TinyUSB Library/src/arduino/ports/rp2040/Adafruit_TinyUSB_rp2040.cpp b/.pio/libdeps/pico/Adafruit TinyUSB Library/src/arduino/ports/rp2040/Adafruit_TinyUSB_rp2040.cpp
index 8af08a4..b0449dd 100644
--- a/.pio/libdeps/pico/Adafruit TinyUSB Library/src/arduino/ports/rp2040/Adafruit_TinyUSB_rp2040.cpp
+++ b/.pio/libdeps/pico/Adafruit TinyUSB Library/src/arduino/ports/rp2040/Adafruit_TinyUSB_rp2040.cpp
@@ -141,13 +141,14 @@ void TinyUSB_Device_Task(void) {
 
 // Debug log with Serial1
 #if CFG_TUSB_DEBUG
+HardwareSerial *TinyUSB_Serial_Debug = nullptr;
 int serial1_printf(const char *__restrict format, ...) {
   char buf[256];
   int len;
   va_list ap;
   va_start(ap, format);
   len = vsnprintf(buf, sizeof(buf), format, ap);
-  Serial1.write(buf);
+  if (TinyUSB_Serial_Debug) TinyUSB_Serial_Debug->write(buf);
   va_end(ap);
   return len;
 }
