diff --git a/src/host/usbh.c b/src/host/usbh.c
index 5775bfa..fab84ad 100644
--- a/src/host/usbh.c
+++ b/src/host/usbh.c
@@ -557,6 +557,7 @@ bool tuh_control_xfer (tuh_xfer_t* xfer)
 
   TU_VERIFY(is_idle);
   const uint8_t rhport = usbh_get_rhport(daddr);
+  osal_task_delay(100);
 
   TU_LOG2("[%u:%u] %s: ", rhport, daddr, xfer->setup->bRequest <= TUSB_REQ_SYNCH_FRAME ? tu_str_std_request[xfer->setup->bRequest] : "Unknown Request");
   TU_LOG2_VAR(xfer->setup);
